### PR TITLE
DOC: Remove deprecated methods from code_checks.sh

### PR DESCRIPTION
**Pull request by shriyakalakata**
- [x] xref #58065
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
Could you fix the merge conflict?

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4915,6 +4915,10 @@ def assign(self, **kwargs) -> DataFrame:
             A new DataFrame with the new columns in addition to
             all the existing columns.
 
+        See Also
+        -------
```
Comment (Aloqeely) [inline pandas/core/frame.py:4919]:
```suggestion
        --------
```
There should be a hyphen under each character in "See Also" (so 8 in total)

Comment (shriyakalakata) [reply pandas/core/frame.py:4919]: Oh I see, thank you for pointing that out!

File: ci/code_checks.sh [hunk 2]
```diff
@@ -81,15 +81,12 @@ if [[ -z "$CHECK" || "$CHECK" == "docstrings" ]]; then
...
         -i "pandas.DataFrame.__iter__ SA01" \
-        -i "pandas.DataFrame.assign SA01" \
         -i "pandas.DataFrame.at_time PR01" \
         -i "pandas.DataFrame.axes SA01" \
-        -i "pandas.DataFrame.bfill SA01" \
         -i "pandas.DataFrame.columns SA01" \
         -i "pandas.DataFrame.copy SA01" \
         -i "pandas.DataFrame.droplevel SA01" \
         -i "pandas.DataFrame.dtypes SA01" \
-        -i "pandas.DataFrame.ffill SA01" \
```
Comment (jorisvandenbossche) [inline ci/code_checks.sh:92]: There is also a Series.bfill/ffill a bit below that has to be removed as well

Comment (shriyakalakata) [reply ci/code_checks.sh:92]: Thank you for pointing that out!

File: pandas/core/frame.py [hunk 3]
```diff
@@ -4924,6 +4924,10 @@ def assign(self, **kwargs) -> DataFrame:
             A new DataFrame with the new columns in addition to
             all the existing columns.
 
+        See Also
+        --------
+        DataFrame.assign : Assign new columns to a DataFrame.
```
[the remark this answers is not in the document]
Comment (shriyakalakata) [reply pandas/core/frame.py:4929]: Should I reference `pandas.DataFrame.index`?

[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/frame.py:4929]: I would suggest

```
        DataFrame.loc : Select a subset of a DataFrame by labels.
        DataFrame.iloc : Select a subset of a DataFrame by positions.
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @shriyakalakata

**Pull request merged**